webhook/webhook.go

```diff
@@ -35,5 +42,43 @@ func main() {
 }
 
 func handleGithubEvent(w http.ResponseWriter, r *http.Request) {
-	// TODO (priyawadhwa@): Add logic to handle a Github event here
+	eventType := r.Header.Get("X-GitHub-Event")
+	if eventType != constants.PullRequestEvent {
+		return
+	}
+	event := new(github.PullRequestEvent)
+	if err := json.NewDecoder(r.Body).Decode(event); err != nil {
+		log.Printf("error decoding pr event: %v", err)
+	}
+	if err := handlePullRequestEvent(event); err != nil {
+		log.Printf("error handling pr event: %v", err)
+	}
+}
+
+func handlePullRequestEvent(event *github.PullRequestEvent) error {
+	// Cleanup any deployments if PR was merged or closed
+	if event.GetAction() == constants.ClosedAction {
+		return kubernetes.CleanupDeployment(event)
+	}
+
+	// Only continue if the docs-modifications label was added
+	if event.GetAction() != constants.LabeledAction {
+		return nil
+	}
+
+	if event.GetPullRequest() == nil {
+		return nil
+	}
+
+	// Make sure pull request is open
+	if event.GetPullRequest().GetMerged() {
```

Comment:
would this for not merged but closed PRs evaluate to `false`?

Reply:
I think so, but I actually don't think we need this check since we first check if a PR's `action="closed"`, which is true for both merged and closed PRs. I'll go ahead and remove it.

Reply:
Isn't `action = "closed"` when it's an actual "Close" event? 
I think we need this check, as we do want to make sure we only deploy for open PRs. 
Here's the (contrived) scenario I'm thinking that this would have gone through:
- a PR is closed  (but not merged!)
- someone puts a "documentation-preview" label on it 
we do not want to build it

Reply:
My point is that I think we need to check for `IsOpen()` or something similar - like `! Merged && ! Closed ` :)

Reply:
for sure, done!